### PR TITLE
ci: abstract adding issues to projects

### PR DESCRIPTION
**Pull request by misiekhardcore**
## Description

This PR is abstracting adding issues to projects

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to #20768

### CODE REVIEW

**Comment by misiekhardcore** [comment]
I'm not sure how to test this.
@cmur2 I have two questions here:
1. will the secrets be passed to that action? I think the answer is yes but better to ask
2. is there any clever way of moving the `if` statement that checks for the matching label to the action instead of keeping it in the workflow?

**Comment by cmur2** [comment]
>     1. Do you have any idea how to test this?

I don't know any good way besides merging to `main` and then creating an issue. Be prepared to revert the change after the merge.

>     2. will the secrets be passed to that action? I think the answer is yes but better to ask

I don't think they will and it might be cleaner anyways to pass them explicitly from the calling workflow to show which secret is actually used. Looking at e.g. https://github.com/camunda/camunda/blob/main/.github/actions/setup-zeebe/action.yml#L25-L33 where they do this way.

>     3. is there any clever way of moving the `if` statement that checks for the matching label to the action instead of keeping it in the workflow?

Not sure but I also don't mind having the `if` statement in the workflow - the refactoring from this PR is a tremendous improvement already 👍🏼

File: .github/actions/add-issue-to-project/action.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                }
+              }
+            }
+          }
+        variables: |
+          owner: "camunda"
+          repo: "camunda"
+          issue: ${{ inputs.issueNumber }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GH_PROJECT_AUTOMATION_TOKEN }}
```
Comment (cmur2) [inline .github/actions/add-issue-to-project/action.yml:37]: Let's add a new input `githubToken` to pass the secret explicitly from the workflow

File: .github/workflows/issues-labeled.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    name: Add ${{ matrix.teams.name }} issue to project if relevant component label added
+    runs-on: ubuntu-latest
+    steps:
+      - name: Add labeled issue to project
+        if: ${{ github.event.label.name == matrix.teams.label }}
+        uses: ./.github/actions/add-labeled-to-project
+        with:
+          projectId: ${{ matrix.teams.projectId }}
+          issueNumber: ${{ github.event.issue.number }}
+          projectAutomationToken: ${{ secrets.GH_PROJECT_AUTOMATION_TOKEN }}
```
Comment (cmur2) [inline .github/workflows/issues-labeled.yml:25]: Input name does not match the name defined in the action

**Review by cmur2 — APPROVED** [review]

**Pull request merged**